projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.spec.ts

```diff
@@ -853,6 +861,136 @@ describe('PreviewsComponent', () => {
         checkPreview(previews[i], previewImages[i], i === 0, DEFAULT_PREVIEW_SIZE);
       }
     });
+
```

Comment:
Addresses: 2. the number of previews oscillate between n (number of requested previews) and n-1/n+1, for example when n=4 or n=5